files/image_config/platform/rc.local

```diff
@@ -183,8 +183,32 @@ for x in "$@"; do
 done
 }
 
+program_baud_rate()
+{
+    if [ -n "$aboot_platform" ]; then
+        platform=$aboot_platform
+    elif [ -n "$onie_platform" ]; then
+        platform=$onie_platform
+    else
+        platform=''
+    fi
+
+    # Default console speed: 9600
+    CONSOLE_SPEED=9600
+
+    CONSOLE_CFG=/usr/share/sonic/device/$platform/installer.conf
+    if [ -f $CONSOLE_CFG ]; then
+        . $CONSOLE_CFG
+    fi
+
+    sed -i "s|\-\-keep\-baud .* %I|\-\-keep\-baud $CONSOLE_SPEED %I|g" /lib/systemd/system/serial-getty@.service
```

Comment:
--keep-baud is using the current baud rate, therefore I am not sure if the getty is trying to do auto-negotiate here. I looked at the stack trace, unfortunately, they are kernel stack trace, it is not pinpoint to where the user space program go hanged. I think it is useful to do strace and gdb on the user space agetty to find out where it is blocked.

Reply:
Removed --keep-baud